src/gt4py/next/iterator/embedded.py

```diff
@@ -1092,11 +1094,15 @@ def remap(self, index_field: common.ConnectivityField | fbuiltins.FieldOffset) -
         raise NotImplementedError()
 
     def restrict(self, item: common.AnyIndexSpec) -> Self:
-        if common.is_absolute_index_sequence(item) and all(common.is_named_index(e) for e in item):  # type: ignore[arg-type] # we don't want to pollute the typing of `is_absolute_index_sequence` for this temporary code # fmt: off
+        if (
+            common.is_absolute_index_sequence(item)  # type: ignore[arg-type] # we don't want to pollute the typing of `is_absolute_index_sequence` for this temporary code # fmt: off
+            and all(isinstance(e, common.NamedIndex) for e in item)
```

Comment:
Why is `is_absolute_index_sequence` needed here instead of just doing?
```suggestion
            isinstance(item, Sequence) and all(isinstance(e, common.NamedIndex) for e in item)
```